### PR TITLE
Custom error validator

### PR DESCRIPTION
**Pull request by cindyyan317**
For the same validator, the error message might be different according to the field location, eg https://xrpl.org/book_offers.html#possible-errors
Add this validator to send out a custom error when validation fails.

### CODE REVIEW

File: src/rpc/common/Validators.h [hunk 1]
```diff
@@ -377,6 +377,36 @@ class IfType final
...
+{
+    Requirement requirement;
+    RPC::Status error;
+
+public:
+    /**
+     * @brief Constructs a validator that calls the given validator "req" and
+     * return customized error "err"
+     */
+    WithCustomError(Requirement const& req, RPC::Status const& err)
```
Comment (godexsoft) [inline src/rpc/common/Validators.h]: I think `Requirement&& req` is a better choice here in combination with `std::forward<Requirement>(req)`.
`err` could be passed by value or `rvalue ref` + `std::move` too. Value is probably better choice here for copy elision.

**Review by godexsoft — CHANGES_REQUESTED** [review]
Small issue found, let's fix and should be good to go :)

File: src/rpc/common/Validators.h (hunk 1 above)
Comment (cindyyan317) [reply src/rpc/common/Validators.h]: Requirement is a class template parameter, so Requirement&& can only reduced as rvalue .
`error: no viable constructor or deduction guide for deduction of template arguments of 'WithCustomError'
         WithCustomError{
         ^
/Users/cyan/code/clio/src/rpc/common/Validators.h:395:5: note: candidate function [with Requirement = RPCng::validation::CustomValidator] not viable: expects an rvalue for 1st argument
    WithCustomError(Requirement&& req, RPC::Status err)`

I will add a lvalue ctor

Comment (cindyyan317) [reply src/rpc/common/Validators.h]: Discuss offline, we will use value and move inside. Because we have some left value validators

**Review by godexsoft — APPROVED** [review]
LGTM :)

**Pull request merged**